examples/dtrace/README.md

```diff
@@ -0,0 +1,131 @@
+# DTrace in PonyC
+
+[DTrace](http://dtrace.org/guide/preface.html) provides a library for dynamic
+tracing of events. This document describes the implementation of DTrace within
+the Pony compiler. In particular, it focuses on how to use and extend the DTrace
+implementation within Pony.
+
+DTrace is only available for MacOS and BSD. For Linux there is an alternative
+called [SystemTap](https://sourceware.org/systemtap/). SystemTap uses the same
+resources as DTrace, but uses different scripts. You will find more information
+on using SystemTap within the Pony compiler [here](../systemtap/README.md).
+
+## Using DTrace scripts for Pony
+
+You can find various example DTrace scripts in `example/dtrace/`. To run these
+scripts, you can use a command in the form of `[script] -c [binary]`. You need
+to use a PonyC compiler compiled with the DTrace flag to use DTrace. To compile
+a DTrace enabled compiler, you can use the following command: `make
+config=release use=dtrace`.
+
+We can also execute a script using the dtrace command. This is necessary if the
+script does not contain a "Shebang". We can do it using the following command:
+`dtrace -s [script] -c [binary]`.
+
+### MacOS 10.11+
+
+MacOS El Capitan for the first time included "System Integrity Protection". This
+feature does not allow for all DTrace features to be used. This includes custom
+DTrace providers. To get around this problem, the following steps can be
+followed:
+
+1. Boot your Mac into Recovery Mode.
+2. Open the Terminal from the Utilities menu.
+3. Then use the following commands:
+
+```bash
+
+csrutil clear # restore the default configuration first
+
+csrutil enable --without dtrace # disable dtrace restrictions *only*
+
+```
+
+After completing these steps your Mac should be able to use DTrace after a
+restart. For more information visit the following article:
+[link](http://internals.exposed/blog/dtrace-vs-sip.html).
+
+## Writing scripts for DTrace in Pony
+
+The following paragraph will inform you on the Pony provider and it's probes for
+DTrace. We refer to the [DTrace
+documentation](http://dtrace.org/guide/preface.html) for more information on the
+syntax and possibilities.
+
+The Pony provider for DTrace consists of DTrace probes implemented in the
+Runtime. A binary compiled with DTrace enabled will allow the user access to the
+information of the probes. You can find a list of all probes and their arguments
+in [`src/common/dtrace_probes.d`](../../src/common/dtrace_probes.d).  The
+following is a toy example DTrace script:
+
+```
+
+pony$target:::gc-start
+{
+  @ = count();
+}
+
+END
+{
+  printa(@);
+}
+
+```
+
+This script increases a counter every time the "GC Start" probe is *fired* and
+prints it result at the end. Note the way in which we (describe the probe). It
```

Comment:
The "Note the way..." seems unfinished, I think the rest of the paragraph does a good job explaining what happens.